src/Components/WebAssembly/WebAssembly.Authentication/src/Interop/AuthenticationService.ts

```diff
@@ -63,12 +63,32 @@ export interface AuthorizeService {
 
 class OidcAuthorizeService implements AuthorizeService {
     private _userManager: UserManager;
-
+    private _intialSilentSignIn: Promise<void> | undefined;
     constructor(userManager: UserManager) {
         this._userManager = userManager;
     }
 
+    async trySilentSignIn() {
+        if (!this._intialSilentSignIn) {
+            this._intialSilentSignIn = (async () => {
+                try {
+                    await this._userManager.signinSilent();
+                    return;
+                } catch (e) {
+                }
+            })();
+        }
+
+        return this._intialSilentSignIn;
+    }
+
     async getUser() {
+        if (window.parent == window && !window.opener && !window.frameElement &&
+            !location.href.startsWith(this._userManager.settings.redirect_uri!)) {
+            // If we are not inside a hidden iframe, try authenticating silently.
+            await AuthenticationService.instance.trySilentSignIn();
```

Comment:
What are the consequences of doing this unconditionally?

Reply:
The underlying auth implementation uses a hidden iframe to perform multiple "silent" operations. We don't want to spam requests to the server in case the Blazor app gets loaded within a hidden iframe.